electron/src/AutoPowerSwitching.ts

```diff
@@ -51,41 +61,54 @@ export const initSwitch = async (state: 'battery' | 'ac') => {
 
 export const initF5Switch = async () => {
 	let config = getConfig();
-	if (!config.f5Switch || !config.f5Switch.enabled || !config.f5Switch.f5Plans) {
+	if (
+		!config.f5Switch ||
+		!config.f5Switch.enabled ||
+		!config.f5Switch.f5Plans
+	) {
 		return;
 	}
 	const plan_names: string[] = config.f5Switch.f5Plans;
-	let old_plan_name = current_plan
-	let new_plan_name = undefined
-	let new_plan = undefined
-
-	for (var x = 0; x < plan_names.length; x++) {
-		if (plan_names[x] === current_plan || x === plan_names.length-1){
-			new_plan_name = plan_names[(x+1) % plan_names.length]
-			break
-		}
+	let plan_index = config.f5Switch.index;
+	if (!isNumber(plan_index) || plan_index > plan_names.length) {
+		plan_index = 0;
+	} else {
+		plan_index = plan_index + 1;
```

Comment:
Maybe we could do the modulo here
```suggestion
		plan_index = (plan_index + 1) % plan_names.length;
```
so that later we store a valid index value in the config (right now it could be stored while equal to `plan_names.length` which seems dirty)

Reply:
Well a plan name can be, as you say, deleted, and it's possible there will be two plan names that are not unique (which would be dumb, and I think I disallowed that, but it's still a possible error).  Having an index means that you're guaranteed to always have the next plan in the list. I think when I tried your method, I had an issue where my first plan was applied two or three times before it started cycling. I guess I just like the idea of an index, and an index that doesn't get unreasonably large (for example someone powerswitches 842 times and the index is now (842 + 1) % plan_names.length. Having the index be something like "1" is so simple and easy to understand for someone who is looking at the config file.

Reply:
IMO the most robust thing would be to have a uuid per power plan, hidden from the user, as Windows does. Sure deleting one would mean some more code to handle this case, but the index solution will just assume that you're at the "next plan" when you're not. 
e.g. this scenario:
```
 Existing Plans: [A,B,C,D,E]
 AC Plan: A, DC Plan: E
 F5 Switch Plans: [A,C,D]
 
 - User presses F5 -> Applies A
 - User presses F5 -> Applies C
 - User unplugs and replugs laptop -> Applies E then A
 - User press F5 -> Applies D
```
the last change is a bit unpredictable for the user, as we don't track all the plan changes.

We could store the applied plan's uuid at a more globlal level in the config, no matter the feature used to apply it. This would give this scenario instead:
```
 Existing Plans: [A,B,C,D,E]
 AC Plan: A, DC Plan: E
 F5 Switch Plans: [A,C,D]
 
 - User presses F5 -> Applies A
 - User presses F5 -> Applies C
 - User unplugs and replugs laptop -> Applies E then A
 - User press F5 -> Applies C
```
which seems more logical to me as you know for sure that when you're on plan A, no matter how you applied it, pressing F5 will always apply plan C next

Reply:
That makes sense to me. I think it could be solved by having a function that edits the index value and writes it, if the plan being switched to is in the list of plans, which I guess is similar to your previous method, but with a persistent index value. I think adding a GUID, which would be possible, and superior, is possible but also a lot of work since we would need to figure out how to apply a guid to all current plans *only* if they don't currently have a guid for the first startup with the new update.